File: backend/internal/models/ioc.go
package models

import "time"

// IOCType catégorise le type d'indicateur de compromission
type IOCType string

const (
	IOCTypeIP     IOCType = "ip"
	IOCTypeDomain IOCType = "domain"
	IOCTypeHash   IOCType = "hash"
	IOCTypeURL    IOCType = "url"
	IOCTypeEmail  IOCType = "email"
	IOCTypeCIDR   IOCType = "cidr" // Ajout BGP Lookup
)

// IOCTLP represente le niveau de sensibilite TLP
type IOCTLP string

const (
	TLPWhite IOCTLP = "white"
	TLPGreen IOCTLP = "green"
	TLPAmber IOCTLP = "amber"
	TLPRed   IOCTLP = "red"
)

// IOCStatus represente l'etat de l'indicateur
type IOCStatus string

const (
	IOCStatusActive        IOCStatus = "active"
	IOCStatusArchived      IOCStatus = "archived"
	IOCStatusFalsePositive IOCStatus = "false_positive"
)

// IOC - Indicateur de Compromission
type IOC struct {
	ID          uint      `json:"id"            gorm:"primaryKey;autoIncrement"`
	Type        IOCType   `json:"type"          gorm:"type:text;not null;index"`
	Value       string    `json:"value"         gorm:"type:text;not null;index"`
	Source      string    `json:"source"        gorm:"type:text"`
	TLP         IOCTLP    `json:"tlp"           gorm:"type:text;not null;default:'white'"`
	Status      IOCStatus `json:"status"        gorm:"type:text;not null;default:'active';index"`
	Tags        string    `json:"tags"          gorm:"type:text"`
	Notes       string    `json:"notes"         gorm:"type:text"`
	MitreTechID string    `json:"mitre_tech_id" gorm:"type:text;index"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

File: frontend/src/types/ioc.ts
// 'cidr' ajouté pour les préfixes réseau exportés depuis le module BGP Lookup
export type IOCType = 'ip' | 'domain' | 'hash' | 'url' | 'email' | 'cidr'

export type IOCTLP = 'white' | 'green' | 'amber' | 'red'

export type IOCStatus = 'active' | 'archived' | 'false_positive'

export interface IOC {
  id: number
  type: IOCType
  value: string
  source: string
  tlp: IOCTLP
  status: IOCStatus
  tags: string       // JSON array encodé en string ex: '["ransomware","c2"]'
  notes: string
  mitre_tech_id: string
  created_at: string
  updated_at: string
}

export interface IOCCreatePayload {
  type: IOCType
  value: string
  source?: string
  tlp?: IOCTLP
  status?: IOCStatus
  tags?: string
  notes?: string
  mitre_tech_id?: string
}

export interface IOCUpdatePayload {
  type?: IOCType
  value?: string
  source?: string
  tlp?: IOCTLP
  status?: IOCStatus
  tags?: string
  notes?: string
  mitre_tech_id?: string
}

export interface IOCListResponse {
  items: IOC[]
  total: number
  page: number
  limit: number
}

export interface IOCStats {
  total: number
  by_type: Partial<Record<IOCType, number>>
}

export interface IOCFilter {
  type?: IOCType | ''
  status?: IOCStatus | ''
  tlp?: IOCTLP | ''
  q?: string
  page?: number
  limit?: number
}
